lib/pricing-poller.js

```diff
@@ -136,44 +136,49 @@ class PricingPoller {
 
     for (let region of this.regions) {
       let ec2 = this.ec2[region];
-      let [azResult, priceResult] = await Promise.all([
-        this.runaws(ec2, 'describeAvailabilityZones', {
-          Filters: [{
-            Name: 'state',
-            Values: ['available'],
-          }],
-        }),
-        this.runaws(ec2, 'describeSpotPriceHistory', {
+      let nextToken = undefined;
+      let azResult = await this.runaws(ec2, 'describeAvailabilityZones', {
+        Filters: [{
+          Name: 'state',
+          Values: ['available'],
+        }],
+      });
+      let zones = azResult.AvailabilityZones.map(x => x.ZoneName);
+
+      do {
+        let priceResult = await this.runaws(ec2, 'describeSpotPriceHistory', {
           StartTime: startTime,
+          NextToken: nextToken,
```

Comment:
I know it's probably paranoia, but in this code base I generally create an object and only the properties that are needed.  This is because a defined property with a value of undefined is different to a property that never has been defined, but returns undefined when dereferenced...  The JS object model is fun.

```js
> a = {}
{}
> a.b=undefined
undefined
> a
{ b: undefined }
>
> delete a.b
true
> a
{}
```